test/src/e2e.dynval/dv.n+1.test.ts

```diff
@@ -0,0 +1,265 @@
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import * as chai from "chai";
+import { expect } from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+import { SDK } from "codechain-sdk";
+import * as stake from "codechain-stakeholder-sdk";
+import "mocha";
+
+import { validators as originalValidators } from "../../tendermint.dynval/constants";
+import { faucetAddress, faucetSecret } from "../helper/constants";
+import { PromiseExpect } from "../helper/promise";
+import { withNodes } from "./setup";
+import CodeChain from "../helper/spawn";
+
+async function getPossibleAuthors(
+    sdk: SDK,
+    blockNumber: number
+): Promise<null | string[]> {
+    return new Promise((resolve, reject) => {
+        sdk.rpc
+            .sendRpcRequest("chain_getPossibleAuthors", [blockNumber])
+            .then(result => {
+                if (
+                    result === null ||
+                    result.every((str: any) => typeof str === "string")
+                ) {
+                    resolve(result);
+                }
+                return reject(
+                    Error(
+                        `Expected chain_getPossibleAuthors to return an array of string or null but it returned ${result}`
+                    )
+                );
+            })
+            .catch(reject);
+    });
+}
+
+chai.use(chaiAsPromised);
+
+const [betty, , , ...otherDynValidators] = originalValidators;
+const allDynValidators = [betty, ...otherDynValidators];
+
+describe("Dynamic Validator N -> N+1", function() {
+    this.timeout(0);
+
+    const promiseExpect = new PromiseExpect();
+    const termSeconds = 20;
+
+    async function beforeInsertionCheck(checkingNode: CodeChain) {
+        const blockNumber = await checkingNode.sdk.rpc.chain.getBestBlockNumber();
+        const termMedata = await stake.getTermMetadata(
+            checkingNode.sdk,
+            blockNumber
+        );
+        const currentTermInitialBlockNumber =
+            termMedata!.lastTermFinishedBlockNumber + 1;
+        const validatorsBefore = await getPossibleAuthors(
+            checkingNode.sdk,
+            currentTermInitialBlockNumber
+        );
+
+        expect(termMedata!.currentTermId).to.be.equals(1);
+        expect(validatorsBefore!.length).to.be.equals(
+            otherDynValidators.length
+        );
+        expect(validatorsBefore).contains.all.members(
+            otherDynValidators.map(validator => validator.platformAddress.value)
+        );
```

Comment:
Let's check Betty is missing in the list.
The `contains.all` function does not guarantee that Betty is not in the list.

Reply:
I thought that the above length check and `contains.all` check guarantees the set identity between `validatorsBefore` and `otherDynValidators`. And `otherDynValidators` does not contain `betty`.
Is it too implicit?

Reply:
I added betty check explicitly